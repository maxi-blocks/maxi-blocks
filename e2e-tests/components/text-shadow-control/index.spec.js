/* eslint-disable no-await-in-loop */
/**
 * WordPress dependencies
 */
import { createNewPost, pressKeyWithModifier } from '@wordpress/e2e-test-utils';
/**
 * Internal dependencies
 */
import {
	getAttributes,
	openSidebarTab,
	getBlockStyle,
	changeResponsive,
	insertMaxiBlock,
	updateAllBlockUniqueIds,
} from '../../utils';

describe('TextShadowControl', () => {
	it('Checking the text shadow control', async () => {
		await createNewPost();
		await insertMaxiBlock(page, 'Text Maxi');
		await updateAllBlockUniqueIds(page);
		await page.keyboard.type('Testing Text Maxi', { delay: 100 });
		const accordionPanel = await openSidebarTab(
			page,
			'style',
			'typography'
		);

		await accordionPanel.$eval(
			'.maxi-tabs-content .maxi-typography-control .maxi-textshadow-control .maxi-toggle-switch .maxi-base-control__label',
			use => use.click()
		);

		await accordionPanel.$$(
			'.maxi-textshadow-control.maxi-typography-control__text-shadow .maxi-default-styles-control'
		);

		const shadowStyles = [
			'none',
			'2px 4px 3px rgba(var(--maxi-light-color-8,150,176,203),0.3)',
			'2px 4px 3px rgba(var(--maxi-light-color-8,150,176,203),0.5)',
			'4px 4px 0px rgba(var(--maxi-light-color-8,150,176,203),0.21)',
		];

		for (let i = 0; i < shadowStyles.length; i += 1) {
			const setting = shadowStyles[i];

			await accordionPanel.$$eval(
				'.maxi-textshadow-control.maxi-typography-control__text-shadow .maxi-default-styles-control button',
				(buttons, i) => buttons[i].click(),
				i
			);
			await page.waitForTimeout(200);

			expect(await getAttributes('text-shadow-general')).toStrictEqual(
				setting
			);
		}

		// change values manually
		const editTextShadow = await page.$$(
			'.maxi-textshadow-control .maxi-advanced-number-control'
		);

		// change X
		await editTextShadow[1].$eval('input', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('34');

		// 	change Y
		await editTextShadow[2].$eval('input', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('12');

		// change Blur
		await editTextShadow[3].$eval('input', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('54');

		expect(await getAttributes('text-shadow-general')).toStrictEqual(
			'34px 12px 54px rgba(var(--maxi-light-color-8,150,176,203),0.21)'
		);

		expect(await getBlockStyle(page)).toMatchSnapshot();
	});

	it('Checking the text shadow control in responsive', async () => {
		await changeResponsive(page, 's');

		// expect values
		// S
		const baseSValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[2].value
		);
		expect(baseSValue).toStrictEqual('34');

		// Y
		const baseYValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[4].value
		);
		expect(baseYValue).toStrictEqual('12');

		// Blur
		const baseBlurValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[6].value
		);
		expect(baseBlurValue).toStrictEqual('54');

		// change values in S responsive
		const editTextShadow = await page.$$(
			'.maxi-textshadow-control .maxi-advanced-number-control'
		);

		// change X
		await editTextShadow[1].$eval('input', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('67');

		// 	change Y
		await editTextShadow[2].$eval('input', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('15');

		// change Blur
		await editTextShadow[3].$eval('input', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('48');

		// expect in Xs
		await changeResponsive(page, 'xs');

		// S
		const xsSValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[2].value
		);
		expect(xsSValue).toStrictEqual('67');

		// Y
		const xsYValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[4].value
		);
		expect(xsYValue).toStrictEqual('15');

		// Blur
		const xsBlurValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[6].value
		);
		expect(xsBlurValue).toStrictEqual('48');

		// expect in M
		await changeResponsive(page, 'm');

		// S
		const mSValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[2].value
		);
		expect(mSValue).toStrictEqual('34');

		// Y
		const mYValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[4].value
		);
		expect(mYValue).toStrictEqual('12');

		// Blur
		const mBlurValue = await page.$$eval(
			'.maxi-textshadow-control .maxi-advanced-number-control input',
			input => input[6].value
		);
		expect(mBlurValue).toStrictEqual('54');
	});
});
