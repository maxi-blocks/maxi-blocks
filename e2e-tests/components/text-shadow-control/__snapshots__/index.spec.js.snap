// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextShadowControl Checking the text shadow control 1`] = `"<style>body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 .maxi-text-block--link, .maxi-text-block--link span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 .maxi-text-block--link, .maxi-text-block--link span{ transition: all 0.3s ease;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content{ color: var(--maxi-light-p-color,rgba(var(--maxi-light-color-3),1)); text-shadow: 4px 4px 0px rgba(var(--maxi-light-color-8),0.21);}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content{ color: var(--maxi-light-link,rgba(var(--maxi-light-color-4),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:hover,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:hover{ color: var(--maxi-light-link-hover,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:hover span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:hover span{ color: var(--maxi-light-link-hover,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:active,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:active{ color: var(--maxi-light-link-active,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:active span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:active span{ color: var(--maxi-light-link-active,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:visited,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:visited{ color: var(--maxi-light-link-visited,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:visited span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 a p.maxi-text-block__content:visited span{ color: var(--maxi-light-link-visited,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a{ color: var(--maxi-light-link,rgba(var(--maxi-light-color-4),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:hover,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:hover{ color: var(--maxi-light-link-hover,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:hover span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:hover span{ color: var(--maxi-light-link-hover,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:active,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:active{ color: var(--maxi-light-link-active,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:active span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:active span{ color: var(--maxi-light-link-active,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:visited,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:visited{ color: var(--maxi-light-link-visited,rgba(var(--maxi-light-color-6),1));}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:visited span,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.text-maxi-12 p.maxi-text-block__content a:visited span{ color: var(--maxi-light-link-visited,rgba(var(--maxi-light-color-6),1));}</style>"`;
